addons/html_builder/static/src/website_blog/author_avatar_many2one_plugin.js

```diff
@@ -0,0 +1,30 @@
+import { patch } from "@web/core/utils/patch";
+import { Many2OneOptionPlugin } from "../website_builder/plugins/options/many2one_option_plugin";
+
+patch(Many2OneOptionPlugin, {
+    dependencies: ["history"],
+});
+
+patch(Many2OneOptionPlugin.prototype, {
+    getActions() {
+        const actions = super.getActions();
+        const superApply = actions.many2OneAction.apply;
+        actions.many2OneAction.apply = (args) => {
+            superApply(args);
+            const { editingElement, value } = args;
+            const { id } = JSON.parse(value);
+            const { oeId, oeField } = editingElement.dataset;
+
+            if (oeField === "author_id") {
+                this.dependencies.history.ignoreDOMMutations(() => {
+                    for (const img of this.document.querySelectorAll(
+                        `[data-oe-model="blog.post"][data-oe-id="${oeId}"][data-oe-field="author_avatar"] img`
+                    )) {
+                        img.src = `/web/image/res.partner/${id}/avatar_1024`;
+                    }
+                });
+            }
+        };
+        return actions;
```

Comment:
I always prefefer to replace the full object.
actions.many2OneAction = {...actions.many2OneAction, apply: () => ...}